src/VisualStudio/Core/Def/Implementation/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -130,11 +130,7 @@ internal Metadata GetMetadata(string fullPath, DateTime snapshotTimestamp)
 
             if (VsSmartScopeCandidate(key.FullPath) && TryCreateAssemblyMetadataFromMetadataImporter(key, out var newMetadata))
             {
-                ValueSource<Optional<AssemblyMetadata>> metadataValueSource = _workspace.Options.GetOption(WorkspaceConfigurationOptions.DisableReferenceManagerWeakRuntimeReferences)
-                    ? new ConstantValueSource<Optional<AssemblyMetadata>>(newMetadata)
-                    : new WeakValueSource<AssemblyMetadata>(newMetadata);
-
-                if (!_metadataCache.GetOrAddMetadata(key, metadataValueSource, out metadata))
+                if (!_metadataCache.GetOrAddMetadata(key, new ConstantValueSource<Optional<AssemblyMetadata>>(newMetadata), out metadata))
```

Comment:
metadata cache still uses RecoverableMetadataValueSource below.  so we need to keep this as a ValueSource.